# Supplementary Tables for Chapter 4 of PhD thesis



## Supplementary Table 13

**Differential alternative splicing events from EventPointer analysis in the Clariom D dataset.**



## Supplementary Table 14

**CancerMine results.**




## Supplementary Table 15

**Differentially expressed alternative splicing events identified in the TCGA-PRAD dataset.**

TCGA: The Cancer Genome Atlas; PRAD: prostate adenocarcinoma; ID: event identifier; logFC: log2 transformed fold change; AveExpr: average expression (log2-transformed) for the event over all samples; t: moderared t-statistic; P.Value: raw p-value; adj.P.Val:adjusted p-value using Benjamini-Hochberg (BH) approach; B: log-odds; gene: parent gene symbol of the event; asType: event type; ID2: alternative identifier; AA: alternate acceptor sites; AD: alternate donor sites; AP: alternate promoter; AT: alternate terminator; ES: exon skip; ME: mutually exclusive exons; RI: retained intron.



## Supplementary Table 16

**rMATS results for PRJEB2449 cohort.**

(a) A summary of the total number of events, significant annotated splicing events, and their associated parent genes as identified using rMATS in the PRJEB2449 dataset. rMATS: Replicate Multivariate Analysis of Transcript Splicing. (b) rMATS output for significant alternative 3' splice site (A3SS) events. rMATS: Replicate Multivariate Analysis of Transcript Splicing. (c) rMATS output for significant alternative 5' splice site (A5SS) events. rMATS: Replicate Multivariate Analysis of Transcript Splicing. (d) rMATS output for significant mutually exclusive exons (MXE) events. rMATS: Replicate Multivariate Analysis of Transcript Splicing. (e) rMATS output for significant retained intron (RI) events. rMATS: Replicate Multivariate Analysis of Transcript Splicing. (f) rMATS output for significant skipped exon (SE) events. rMATS: Replicate Multivariate Analysis of Transcript Splicing.




## Supplementary Table 17

**Overlapping events among the cohorts.**

(a) Overlapping differentially expressed alternative spicing events between the Clariom D and TCGA-PRAD datasets. TCGA: The Cancer Genome Atlas; PRAD: prostate adenocarcinoma. (b) Overlapping differentially expressed alternative spicing events between the Clariom D and PRJEB2449 datasets. (c) Overlapping differentially expressed alternative spicing events across the Clariom D, TCGA-PRAD and PRJEB2449 datasets. TCGA: The Cancer Genome Atlas; PRAD: prostate adenocarcinoma. (d) Overlapping differentially expressed alternative spicing events observed between the TCGA-PRAD and PRJEB2449 datasets. TCGA: The Cancer Genome Atlas; PRAD: prostate adenocarcinoma.



## Supplementary Table 18


**Differentially expressed splicing factors and prognostic alternative splicing events.**

(a) Results of the differential gene expression analysis of the splicing factors in the TCCGA-PRAD dataset. TCGA: The Cancer Genome Atlas; PRAD: prostate adenocarcinoma; logFC: log2 transformed fold change; AveExpr: average expression (log2-transformed) for the gene over all samples; t: moderared t-statistic; P.Value: raw p-value; adj.P.Val:adjusted p-value using Benjamini-Hochberg (BH) approach; B: log-odds. (b) Results of the univariate Cox regression analysis of the biochemical recurrence-free survival (BCRFS)-associated events in the TCGA-PRAD datasets. TCGA: The Cancer Genome Atlas; PRAD: prostate adenocarcinoma; z: the values of the Wald statistic; HR: hazard ratio of the event; HR.95L: the lower limit of 95% confidence interval; HR.95H: the upper limit of the 95% confidence interval; p: p-values obtained from the likelihood ratio test.




## Supplementary Table 19

**Screening for candidate prognostic events associated with BCRFS.**

(a) Univariate Cox regression analysis of overlapping differentially expressed alternative splicing (DEAS) events identified in both The Cancer Genome Atlas-Prostate Adenocarcinoma (TCGA-PRAD) and PRJEB2449 cohorts, as related to the biochemical recurrence-free survival (BCRFS) of prostate cancer (PCa) patients.z: the values of the Wald statistic; HR: hazard ratio of the event; HR.95L: the lower limit of 95% confidence interval; HR.95H: the upper limit of the 95% confidence interval; p: p-values obtained from the likelihood ratio test. (b) Kaplan-Meier analysis of the overlapping differentially expressed alternative splicing (DEAS) events identified in both The Cancer Genome Atlas-Prostate Adenocarcinoma (TCGA-PRAD) and PRJEB2449 cohorts, with regard to the biochemical recurrence-free survival (BCRFS) of prostate cancer (PCa) patients. 'p-value' refers to the statistical significance as determined by the log-rank test. (c) Multivariate Cox regression analysis of the 21 intersected events identified from both the univariate Cox regression and the Kaplan-Meier (KM) analyses, pertaining to the biochemical recurrence-free survival (BCRFS) of prostate cancer (PCa) patients. The coefficients are represented by 'coef'; HR: hazard ratio of the event; HR.95L: the lower limit of 95% confidence interval; HR.95H: the upper limit of the 95% confidence interval; p: p-values obtained from the likelihood ratio test.




## Supplementary Table 20


**Cox regression between our event signature and traditional clinicopathological parameters.**

(a) Univariate and multivariate Cox regression of the alternative splicing event signature and clinicopathological factors with respect to the biochemical recurrence-free survival of prostate cancer patients in the TCGA-PRAD training set. TCGA: The Cancer Genome Atlas; PRAD: prostate adenocarcinoma; Age: age at diagnosis; T stage: tumour stage; N stage: lymph node status (N0 = without lymph node metastasis; N1 = with lymph node metastasis). HR: hazard ratio of the parameter; HR.95L: the lower limit of 95% confidence interval; HR.95H: the upper limit of the 95% confidence interval; p: p-values obtained from the likelihood ratio test. (b) Univariate and multivariate Cox regression of the alternative splicing event signature and clinicopathological factors with respect to the biochemical recurrence-free survival of prostate cancer patients in the TCGA-PRAD testing set. TCGA: The Cancer Genome Atlas; PRAD: prostate adenocarcinoma; Age: age at diagnosis; T stage: tumour stage; N stage: lymph node status (N0 = without lymph node metastasis; N1 = with lymph node metastasis). HR: hazard ratio of the parameter; HR.95L: the lower limit of 95% confidence interval; HR.95H: the upper limit of the 95% confidence interval; p: p-values obtained from the likelihood ratio test. (c) Univariate and multivariate Cox regression of the alternative splicing event signature and clinicopathological factors with respect to the biochemical recurrence-free survival of prostate cancer patients in the TCGA-PRAD complete set. TCGA: The Cancer Genome Atlas; PRAD: prostate adenocarcinoma; Age: age at diagnosis; T stage: tumour stage; N stage: lymph node status (N0 = without lymph node metastasis; N1 = with lymph node metastasis). HR: hazard ratio of the parameter; HR.95L: the lower limit of 95% confidence interval; HR.95H: the upper limit of the 95% confidence interval; p: p-values obtained from the likelihood ratio test.






